enclave/crypto/sha.c

```diff
@@ -77,3 +77,53 @@ oe_result_t oe_sha256_final(oe_sha256_context_t* context, OE_SHA256* sha256)
 done:
     return result;
 }
+
```

Comment:
I've gated them for now. I would have implemented it for BCrypt too, but it exposes only a `void*` and I'd have to guess the memory layout behind it.

Reply:
FWIW, I took a look at what it would take to implement it on the Windows side and I'm not keen on supporting it. BCrypt exposes it as an opaque handle for a reason, because the actual context values looks to be several layers deep and the data structures pointed to by the handler are assembled in a "generic" way that makes me very nervous to take a dependency on the nesting of that closed source data structure being a stable memory layout.

Reply:
Thanks for taking a look at this, that's good to know! I don't know whether BCrypt is a hard dependency because of some features that other libraries don't support. In the worst case, we could always use a different hash implementation (e.g. EverCrypt), but that's another bit of complication of course.